update to activesupport 4.2

Ch4s3 opened the pull request.

stve commented:
any progress on this fellas? In the process of upgrading an app and just realized this is a blocker for me. Thanks.

Ch4s3 commented:
@stve It's a blocker for my team as well, but I/we haven't had time to figure it out yet. The only issue is the Mysql aes search discussed above. If you have any thoughts, we can probably get it knocked out soon. (As long as the problem isn't with ActiveRecord or the query builder)

dwilkie commented on lib/crypt_keeper/provider/mysql_aes_new.rb:
```diff
@@ -36,7 +36,7 @@ def decrypt(value)
       # Returns an Enumerable
       def search(records, field, criteria)
-        records.where(field.to_sym => encrypt(criteria))
+        records.select { |record| record[field] == criteria }
```
https://github.com/Ch4s3/crypt_keeper/pull/1 fixes the issue

dwilkie commented:
https://github.com/Ch4s3/crypt_keeper/pull/1

Ch4s3 commented:
@jmazzi It looks like @dwilkie sorted out the MySql issue. It should be ready to go now.

jmazzi commented:
@Ch4s3 @dwilkie awesome, did you want to open up a PR?

Ch4s3 commented:
@jmazzi This ( #94 ) is the pull request for the update. I merged @dwilkie 's  into support_rails_4_2 branch on Ch4s3/crypt_keeper.

jmazzi commented:
@Ch4s3 nice! can you squash it into one commit and document the change? I will release it right after merging.

Ch4s3 commented:
@jmazzi  Its squashed and I added a bit to the Readme that says tested against 4.2, I wasn't sure what/where else to document the changes. I also added MRI 2.2 to the travis matrix for AR 4.2, since I feel like most people using 4.2 will want to use 2.2.

jmazzi commented on crypt_keeper.gemspec:
```diff
@@ -16,8 +16,8 @@ Gem::Specification.new do |gem|
-  gem.add_runtime_dependency 'activerecord',  '>= 3.1', '< 4.2'
-  gem.add_runtime_dependency 'activesupport', '>= 3.1', '< 4.2'
+  gem.add_runtime_dependency 'activerecord',  '>= 3.1', '<= 4.2'
+  gem.add_runtime_dependency 'activesupport', '>= 3.1', '<= 4.2'
```
@Ch4s3 is there a specific reason you went with `<=` over `< 4.3`? If not, I'm going to change that in master after merging.

Ch4s3 replied:
No specific reason, but I'm not sure if 4.3 will be a thing or if they're jumping straight to 5.0, I haven't been following AR/Rails too closely lately.

jmazzi commented:
This fix was released as 0.19.0

> On Feb 2, 2015, at 3:11 PM, Chase Gilliam <notifications@github.com> wrote:
> 
> @jmazzi <https://github.com/jmazzi> Its squashed and I added a bit to the Readme that says tested against 4.2, I wasn't sure what/where else to document the changes. I also added MRI 2.2 to the travis matrix for AR 4.2, since I feel like most people using 4.2 will want to use 2.2.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jmazzi/crypt_keeper/pull/94#issuecomment-72528468>.
>
